src/pykiso/lib/connectors/cc_pcan_can/cc_pcan_can.py

```diff
@@ -559,10 +569,16 @@ def start_pcan_trace(self, trace_path: Optional[str] = None, trace_size: int = 1
             the trace path defined for the last trace created, defaults to None
         :param trace_size:  maximum size of the trace (in MB), defaults to 10
         """
+        if not self.logging_activated:
+            log.warning("Logging is not activated")
+            return
         if self.trace_running:
             log.warning("Trace is already started")
             return
 
+        if self.raw_pcan_interface is None:
+            self._cc_open()
```

Comment:
The trace should be started only if the channel is open. We also have locks to protect the resources and this is here a bypass.

Reply:
I added this  `cc_open` because if you use the logic with the auxiliary with auto_start to True, then the connection will not be opened then I can't start the  pcan trace automatically.

Reply:
But sorry, if this is a functionality that should be provided with the auto-start, it needs to become a parameter and also started there. As of today, this is a API the user calls and should stay like that.

Can you please provide a fix for that?